pkgs/tools/misc/optinix/default.nix

```diff
@@ -0,0 +1,39 @@
+{
+  lib,
+  fetchFromGitLab,
+  buildGoModule,
+  installShellFiles,
+}:
+buildGoModule rec {
+  pname = "optinix";
+  version = "0.1.1";
+
+  src = fetchFromGitLab {
+    owner = "hmajid2301";
+    repo = "optinix";
+    rev = "v${version}";
+    sha256 = "sha256-bRHesc03jExIL29BCP93cMbx+BOT4sHCu58JjpmRaeA=";
+  };
+
+  vendorHash = "sha256-uSFEhRWvJ83RGpekPJL9MOYJy2NfgVdZxuaNUMq3VaE=";
+
+  nativeBuildInputs = [
+    installShellFiles
+  ];
+
+  postInstall = ''
+    installShellCompletion --cmd optinix \
+      --bash <($out/bin/optinix completion bash) \
+      --fish <($out/bin/optinix completion fish) \
+      --zsh <($out/bin/optinix completion zsh)
+  '';
+
+  meta = with lib; {
+    description = "Tool for searching options in Nix.";
```

Comment:
Thanks for accepting the review suggestion, @hmajid2301, much appreciated, on more it: 🙂 
> meta.description must […] [n]ot end with a period (or any punctuation for that matter).

— https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#meta-attributes
```suggestion
    description = "Tool for searching options in Nix";
```